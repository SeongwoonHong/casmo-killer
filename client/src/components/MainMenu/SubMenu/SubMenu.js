import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import './SubMenu.scss';

class SubMenu extends Component {
  render() {
    return (
      <div className="sub-menu">
        <h3>커뮤니티</h3>
        <ul>
          <li>
            <NavLink to="/">
              &middot; 모든 커뮤니티
            </NavLink>
          </li>
          <li>
            <NavLink to="/">
              &middot; 나의 커뮤니티
            </NavLink>
          </li>
          <li>
            <NavLink to="/">
              &middot; 즐겨찾기
            </NavLink>
          </li>
          <li>
            <NavLink to="/">
              &middot; 자유게시판
            </NavLink>
          </li>
          <li>
            <NavLink to="/">
              &middot; 장터
            </NavLink>
          </li>
        </ul>
      </div>
    );
  }
}

export default SubMenu;
