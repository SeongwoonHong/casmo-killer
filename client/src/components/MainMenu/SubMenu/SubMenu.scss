.sub-menu {
    background-color: #004d40;
    width: 100%;
    > h3 {
        color: #26a69a;
        font-size: 22px;
        line-height: 50px;
        margin: 0;
        padding-left: 30px;
    }
    > ul {
        > li {
            width: 100%;
            > a {
                color: #e0f2f1;
                line-height: normal;
                padding: 10px 25px;
            }
        }
    }
}
