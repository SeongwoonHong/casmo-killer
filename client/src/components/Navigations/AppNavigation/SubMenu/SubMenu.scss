.Sub-menu {
    background-color: palette(Grey, 900);
    display: none;
    @media screen and (min-width: 992px) {
        display: block;
        > ul {
            @include flex-box(flex-start);
            list-style: none;
            margin: 0 auto;
            max-width: 1200px;
            width: 100%;
            > li {
                text-align: center;
                width: 130px;
                > a {
                    border-bottom: 2px solid transparent;
                    color: palette(Grey, 500);
                    display: block;
                    font-size: 12px;
                    padding: 10px 0;
                    text-decoration: none;
                    &.active {
                        border-color: palette(Grey, 100);
                        color: palette(Grey, 100);
                        font-weight: 700;
                    }
                }
            }
        }
    }
}
