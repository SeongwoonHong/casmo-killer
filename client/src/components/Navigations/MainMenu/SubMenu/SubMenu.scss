.sub-menu {
    bottom: 0;
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    width: 220px;
    > h3 {
        font-size: 20px;
        line-height: 50px;
        margin: 0;
        padding: 10px 0 0 25px;
    }
    > ul {
        > li {
            width: 100%;
            > a {
                font-size: 12px;
                line-height: normal;
                padding: 10px 25px;
                > span {
                    font-size: 14px;
                    font-weight: 700;
                    margin-right: 10px;
                }
                &:hover,
                &.active {
                    background-color: rgba(0, 0, 0, 0.1);
                    color: #fff !important;
                }
            }
        }
    }
    @media screen and (min-width: 360px) {
        width: 260px;
    }
    @media screen and (min-width: 992px) {
        width: 170px;
    }
}
