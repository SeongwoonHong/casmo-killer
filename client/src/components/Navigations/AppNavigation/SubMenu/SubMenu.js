import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import './SubMenu.scss';

class SubMenu extends Component {

  render() {

    const { parent, items } = this.props;

    const MenuLinks = (routes) => {
      return routes.map((route) => {
        return (
          <li
            className="Sub-menu__list__items"
            key={ route.name }>
            <NavLink
              className="Sub-menu__list__items__links"
              exact
              to={ parent + route.path }>
              { route.name }
            </NavLink>
          </li>
        );
      });
    };

    return (
      <div className="Sub-menu">
        <ul className="Sub-menu__list component-row">
          { MenuLinks(items) }
        </ul>
      </div>
    );

  }

}

export default SubMenu;
