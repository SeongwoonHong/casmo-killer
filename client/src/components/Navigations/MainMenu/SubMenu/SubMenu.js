import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import './SubMenu.scss';

class SubMenu extends Component {

  render() {

    const { match, title, items } = this.props;

    return (
      <div className="teal darken-4 sub-menu">
        <h3 className="teal-text text-lighten-1">{ title }</h3>
        <ul>
          {
            items
              ? items.map((item) => {
                return (
                  <li key={ item.name }>
                    <NavLink
                      exact
                      to={ match.path + item.path }
                      className="teal-text text-lighten-5">
                      <span>&#183;</span>
                      { item.name }
                    </NavLink>
                  </li>
                );
              })
              : null
          }
        </ul>
      </div>
    );

  }

}

export default SubMenu;
