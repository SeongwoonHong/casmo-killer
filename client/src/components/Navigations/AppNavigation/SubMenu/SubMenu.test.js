import React from 'react';
import { shallow } from 'enzyme';

import SubMenu from './index';

describe('<SubMenu />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(
      <SubMenu items={ [
        {
          name: 'a',
          path: ''
        },
        {
          name: 'b',
          path: ''
        },
        {
          name: 'c',
          path: ''
        }
      ] } />
    );
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();
    expect(component.find('.Sub-menu__list__items').length).toEqual(3);

  });

});
