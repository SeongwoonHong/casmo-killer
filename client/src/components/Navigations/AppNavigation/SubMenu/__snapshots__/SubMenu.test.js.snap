// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubMenu /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubMenu
    items={
        Array [
            Object {
              "name": "a",
              "path": "",
            },
            Object {
              "name": "b",
              "path": "",
            },
            Object {
              "name": "c",
              "path": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="Sub-menu__list component-row"
>
        <li
                className="Sub-menu__list__items"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Sub-menu__list__items__links"
                        exact={true}
                        to="undefined"
                >
                        a
                </NavLink>
        </li>
        <li
                className="Sub-menu__list__items"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Sub-menu__list__items__links"
                        exact={true}
                        to="undefined"
                >
                        b
                </NavLink>
        </li>
        <li
                className="Sub-menu__list__items"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Sub-menu__list__items__links"
                        exact={true}
                        to="undefined"
                >
                        c
                </NavLink>
        </li>
</ul>,
      "className": "Sub-menu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="Sub-menu__list__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Sub-menu__list__items__links"
                        exact={true}
                        to="undefined"
            >
                        a
            </NavLink>
</li>,
          <li
            className="Sub-menu__list__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Sub-menu__list__items__links"
                        exact={true}
                        to="undefined"
            >
                        b
            </NavLink>
</li>,
          <li
            className="Sub-menu__list__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Sub-menu__list__items__links"
                        exact={true}
                        to="undefined"
            >
                        c
            </NavLink>
</li>,
        ],
        "className": "Sub-menu__list component-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="Sub-menu__list__items__links"
              exact={true}
              to="undefined"
>
              a
</NavLink>,
            "className": "Sub-menu__list__items",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "a",
              "className": "Sub-menu__list__items__links",
              "exact": true,
              "to": "undefined",
            },
            "ref": null,
            "rendered": "a",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="Sub-menu__list__items__links"
              exact={true}
              to="undefined"
>
              b
</NavLink>,
            "className": "Sub-menu__list__items",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "b",
              "className": "Sub-menu__list__items__links",
              "exact": true,
              "to": "undefined",
            },
            "ref": null,
            "rendered": "b",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "c",
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="Sub-menu__list__items__links"
              exact={true}
              to="undefined"
>
              c
</NavLink>,
            "className": "Sub-menu__list__items",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "c",
              "className": "Sub-menu__list__items__links",
              "exact": true,
              "to": "undefined",
            },
            "ref": null,
            "rendered": "c",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="Sub-menu__list component-row"
>
          <li
                    className="Sub-menu__list__items"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="Sub-menu__list__items__links"
                              exact={true}
                              to="undefined"
                    >
                              a
                    </NavLink>
          </li>
          <li
                    className="Sub-menu__list__items"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="Sub-menu__list__items__links"
                              exact={true}
                              to="undefined"
                    >
                              b
                    </NavLink>
          </li>
          <li
                    className="Sub-menu__list__items"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="Sub-menu__list__items__links"
                              exact={true}
                              to="undefined"
                    >
                              c
                    </NavLink>
          </li>
</ul>,
        "className": "Sub-menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="Sub-menu__list__items"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="Sub-menu__list__items__links"
                            exact={true}
                            to="undefined"
              >
                            a
              </NavLink>
</li>,
            <li
              className="Sub-menu__list__items"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="Sub-menu__list__items__links"
                            exact={true}
                            to="undefined"
              >
                            b
              </NavLink>
</li>,
            <li
              className="Sub-menu__list__items"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="Sub-menu__list__items__links"
                            exact={true}
                            to="undefined"
              >
                            c
              </NavLink>
</li>,
          ],
          "className": "Sub-menu__list component-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "a",
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="Sub-menu__list__items__links"
                exact={true}
                to="undefined"
>
                a
</NavLink>,
              "className": "Sub-menu__list__items",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "a",
                "className": "Sub-menu__list__items__links",
                "exact": true,
                "to": "undefined",
              },
              "ref": null,
              "rendered": "a",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "b",
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="Sub-menu__list__items__links"
                exact={true}
                to="undefined"
>
                b
</NavLink>,
              "className": "Sub-menu__list__items",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "b",
                "className": "Sub-menu__list__items__links",
                "exact": true,
                "to": "undefined",
              },
              "ref": null,
              "rendered": "b",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "c",
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="Sub-menu__list__items__links"
                exact={true}
                to="undefined"
>
                c
</NavLink>,
              "className": "Sub-menu__list__items",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "c",
                "className": "Sub-menu__list__items__links",
                "exact": true,
                "to": "undefined",
              },
              "ref": null,
              "rendered": "c",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
